pybandits/cmab_meta_model.py

```
@@ -0,0 +1,423 @@
+
+    @model_validator(mode="after")
+    def _check_models(self) -> "CmabMetaModel":
+        """Cross-arm consistency: every head must share input size / update method / update kwargs.
+
+        Adapted from the old ``CmabPerActionMetaModel.check_models``; the heads' input dim must also
+        match the backbone's ``embedding_dim`` when a backbone is present.
+        """
+        if not self.actions:
+            raise AttributeError("At least one action should be defined.")
+        if len(self.actions) == 1:
+            warnings.warn("Only a single action was supplied. This MAB will be deterministic.")
```

Ruff (0.15.17)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)